### ISSUE TITLE
Affirm commit_2pc order 

### ISSUE THREAD

**Issue by rosik** [tarantool/cartridge#818]
Nowadays two-phase commit applies config in parallel on all cluster instances. Sometimes it causes problems, such as https://github.com/tarantool/vshard/issues/237. And I guess other roles may be affected too.

My proposal is to make parallel apply in two waves: first wave runs on leaders, the second wave updates all others.

Q: What if there's no leader? It may be the case in stateful mode.
A: All instances will be updated in the second wave.

Q: Suppose I have a replicaset of two instances without any roles. I'm going to switch the leader and enable storage role. What is the perfect sequence of actions to accomplish this task?
A: ???

**Comment by rosik** [comment]
This issue won't fix anything unless we add some kind of inter-instance synchronization. But it doesn't seem to be appropriate for a distributed system like the Cartridge. Closing.
